Digitall.Warehouse.Application/Abstractions/Persistence/IUnitOfWork.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Digitall.Warehouse.Application.Abstractions.Persistence
+{
+    public interface IUnitOfWork
+    {
+        Task SaveChangesAsync(CancellationToken cancellationToken = default);
```

Comment:
You should add this  ICategoryRepository Products { get; } here and inject it in the implementation. (I wrote more detailed explanation in the implementation class comment)

Reply:
Got you ;) 3 times for good luck :)